.circleci/config.yml

```diff
@@ -1,21 +1,73 @@
-version: 2
+version: 2.1
+
+orbs:
+  node: circleci/node@4.4.0
+
+workflows:
+  version: 2
+  workflow:
+    jobs:
+      - build:
+          matrix:
+            parameters:
+              # We should test against "latest Node 16", but there's an issue with 16.2
+              # This will be updated soon, along with a bug report to Node.js
+              # node-version: ["lts/erbium", "lts/fermium", "16"]
+              node-version: ["lts/erbium", "lts/fermium", "16.1"]
+      - release:
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - main
+
 jobs:
   build:
-    docker:
-      - image: circleci/node:14.8.0
+    parameters:
+      node-version:
+        description: Node version to run tests under (set by matrix build config)
+        type: string
+    machine:
+      image: ubuntu-2004:202104-01
     steps:
       - checkout
-      - setup_remote_docker
-      - run: |
-          cd packages/integration-tests
-          docker-compose build db tests
-          docker-compose up db-wait
-          docker-compose up -d tests
-      - run: |
-          cd packages/integration-tests
-          docker-compose exec tests ./tests.sh
-      - run: |
```

Comment:
This release step is now a separate workflow that depends on `build`. It accomplishes the same thing, it just leans on CircleCI a bit more than this bash `if` statement.

Reply:
Fwiw I kinda liked the bash `if` because it's faster (handwaving on how much faster) than doing the install build/compile again in a new job, but :shrug: not a big deal.

Reply:
Ah, I'm glad you called this out. I assumed the `publish` step would run the build and whatnot, but I'm not sure of that. I went ahead and restored this as part of the `build` step with a conditional block. See [`1996b16` (#121)](https://github.com/stephenh/joist-ts/pull/121/commits/1996b1659fe5aa8c830a049e7bb83e2ac26e480e)